Fixes

phibue opened the pull request:
Channel 2 Color: 
   case of "Gauge": there was a bug
   case of "Waveform": was missing
properties "-p all" was described in help, but not yet implemented

phibue committed:
Merge pull request #2 from MMMZZZZ/master

update

phibue committed:
Gauge & Waveform: "Channel 2 Color", properties "-p all" added

MMMZZZZ committed:
Removed -p all

Was a leftover in the description, shouldn't have been there and makes no sense because of the different unknown* variants. Use "-p visuals unknowns" instead.

MMMZZZZ commented:
Hi,

Thanks for the PR. The `-p all` command was a leftover that became obsolete when I added the different `unknown_*` options. Since you can add any combination of `visuals` and `unknown` keywords, there's no need for the ambiguous `all` keyword anymore. 

Considering the Gauge/Waveform stuff there's definetly something wrong there. Gauge has no channels f.ex. I think I mixed them up. That aside, I don't quite understand your changes to the channel colors. I will have another look at it later.

Kind regards,
Max

phibue commented:
Hi Max

Just a short response:
all: 
Yes, of course, it is not needed. In case of additional future -p options it could become usefull. I assumed you wanted to keep the «all» therefore.

Gauge:
I didn’t check Gauge ( I assumed the intelligent Modules support several Colors somehow)

Waveform:
«Channel 2 Color» was ignored allways. The reason was:
• On "pco2" you initialize  ignore = true, and if type==122 then you set ignore = true again. It never became false. It’s obvious when you see it. (You probably mixed «positive» with «negative» logic.)
• On "pco3": No need to change this, I changed it just for fun. In order to have the same logic as in "pco2"; and it is shorter.
Again: Great work!
Kind regards,
Philipp

Von: MMMZZZZ
Gesendet: Dienstag, 8. Dezember 2020 13:19
An: MMMZZZZ/Nextion2Text
Cc: phibue; Author
Betreff: Re: [MMMZZZZ/Nextion2Text] Fixes (#6)

Hi,
Thanks for the PR. The -p all command was a leftover that became obsolete when I added the different unknown_* options. Since you can add any combination of visuals and unknown keywords, there's no need for the ambiguous all keyword anymore.
Considering the Gauge/Waveform stuff there's definetly something wrong there. Gauge has no channels f.ex. I think I mixed them up. That aside, I don't quite understand your changes to the channel colors. I will have another look at it later.
Kind regards,
Max
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

MMMZZZZ commented:
Ah your description makes sense. While powerful, the dictionary structure also allows many different ways of describing the same thing. So yeah, sounds very plausible that I mixed things up. In the commit diff I didn't see the default `"ignore": true`, hence my confusion. 

As I said I found a couple other issues, I'll push them together with what you found.

MMMZZZZ committed:
Gauge, Waveform fixes, missing vis: True, sendkey no more ignored

MMMZZZZ committed:
Removed "initial..." Takes space, but doesnt give much value

MMMZZZZ committed:
Type comments

MMMZZZZ committed:
some warnings fixed

phibue committed:
correction in -help: visual unknown (not visuals unknowns)

MMMZZZZ committed:
Added license note

MMMZZZZ committed:
typo

MMMZZZZ committed:
Dependency resolver fixed

MMMZZZZ committed:
Renaming for consistency

MMMZZZZ committed:
Added sendkey mapping

MMMZZZZ committed:
Comment for mapping overwrite

MMMZZZZ committed:
Uniformed mapping casing

MMMZZZZ committed:
Removed debugging message

MMMZZZZ committed:
Added Options to keep original attribute names and/or values
